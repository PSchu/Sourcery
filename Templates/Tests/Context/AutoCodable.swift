//
//  AutoCodable.swift
//  TemplatesTests
//
//  Created by Ilya Puchka on 13/04/2018.
//  Copyright Â© 2018 Pixle. All rights reserved.
//

import Foundation

protocol AutoDecodable: Swift.Decodable {}
protocol AutoEncodable: Swift.Encodable {}
protocol AutoCodable: AutoDecodable, AutoEncodable {}

struct CustomKeyDecodableStruct: AutoDecodable {
    let string: String
    let bool: Bool
    let int: Int

    enum CodingKeys: String, CodingKey {
        case int = "integer"

// sourcery:inline:auto:CustomKeyDecodableStruct.CodingKeys.AutoCodable
        // following keys are added automatically by Sourcery
        case string
        case bool
// sourcery:end
}

}
