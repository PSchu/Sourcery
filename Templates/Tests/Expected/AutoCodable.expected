// Generated using Sourcery 0.12.0 â€” https://github.com/krzysztofzablocki/Sourcery
// DO NOT EDIT





extension CustomMethodsCodableStruct {

    enum CodingKeys: String, CodingKey {
        case boolValue
        case intValue
        case optionalString
        case requiredString
        case requiredStringWithDefault
        case computedPropertyToEncode
    }

    public init(from decoder: Decoder) throws {
        let container = try decoder.container(keyedBy: CodingKeys.self)

        boolValue = try CustomMethodsCodableStruct.decodeBoolValue(from: decoder)
        intValue = CustomMethodsCodableStruct.decodeIntValue(from: container) ?? CustomMethodsCodableStruct.defaultIntValue
        optionalString = try container.decodeIfPresent(String.self, forKey: .optionalString)
        requiredString = try container.decode(String.self, forKey: .requiredString)
        requiredStringWithDefault = (try? container.decode(String.self, forKey: .requiredStringWithDefault)) ?? CustomMethodsCodableStruct.defaultRequiredStringWithDefault
    }

    public func encode(to encoder: Encoder) throws {
        var container = encoder.container(keyedBy: CodingKeys.self)

        try encodeBoolValue(to: encoder)
        encodeIntValue(to: &container)
        try container.encodeIfPresent(optionalString, forKey: .optionalString)
        try container.encode(requiredString, forKey: .requiredString)
        try container.encode(requiredStringWithDefault, forKey: .requiredStringWithDefault)
        encodeComputedPropertyToEncode(to: &container)
        try encodeAdditionalVariables(to: encoder)
    }

}
